### ISSUE TITLE
Could not load type 'SharpDX.Bool' from assembly 'SharpDX, Version=3.1.1.0

### ISSUE THREAD

**Issue by przemo1983** [accord-net/framework#448]
There is no SharpDX.Bool in SharpDX, Version=3.1.1.0.
Accord.DirectShow version 3.4.0.

Is this issue in Accord lib or SharpDX?

**Comment by blaisexen** [comment]
There is a SharpDX.Bool in 3.4.0

**Comment by przemo1983** [comment]
This is very strange.
I get following exception for these assemblies:

> An unhandled exception of type 'System.TypeLoadException' occurred in mscorlib.dll
> 
> Additional information: Could not load type 'SharpDX.Bool' from assembly 'SharpDX, Version=3.1.1.0, Culture=neutral, PublicKeyToken=b4dcf0f35e5521f1'.

From 

> Source "Accord.Audio.DirectSound"

> 		StackTrace	"   at Accord.DirectSound.AudioCaptureDevice.WorkerThread()\r\n   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)\r\n   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)\r\n   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)\r\n   at System.Threading.ThreadHelper.ThreadStart()"	

I will try to reinstall all Accord libs.
Please note that I am trying to migrate Audio Recorder sample to my code. Audio recorder sample is run with other versions of assemblies (SharpDX version 2.6).

**Comment by przemo1983** [comment]
I updated SharpDX assemblies in original Audio Recorder sample and in my case I have exception at attempt to make a recording:

> Additional information: Could not load file or assembly 'SharpDX.DirectSound, Version=3.1.1.0, Culture=neutral, PublicKeyToken=b4dcf0f35e5521f1' or one of its dependencies. The system cannot find the file specified.

Could you please to install the latest libraries for Accord.net and SharpDX and try to run Audio Recorder sample app?

**Comment by przemo1983** [comment]
Could you please give me any clue how to resolve this?
If there is an issue with SharpDX library itself I will create issue in that project.

When I reverted SharpDX and SharpDX.DirectSound versions to 2.6 an Audio Recorder sample started to work again (this sample uses 3.4.1-alpha Accord libraries).

Please see attached screenshots.
There is no SharpDX.Bool for 3.1.1. 
![sharpdx_3 1 1](https://cloud.githubusercontent.com/assets/2435251/23252238/705da6ee-f9b0-11e6-8daf-9186dc28414a.png)

There is a SharpDx.Bool for SharpDX 2.6.
![sharpdx_2 6 0](https://cloud.githubusercontent.com/assets/2435251/23252225/5d997182-f9b0-11e6-9483-b2a65b98915c.png)

**Comment by blaisexen** [comment]
Hi, 
I used Accord.Net 3.4.0 with SharpDX 2.6.0 and I found no problem about it.

**Comment by przemo1983** [comment]
Thanks. For your help.

**Comment by fsteveb** [comment]
Go to your project reference and see where the path to the dll is.
Check that folder if the dll is there. You will get that error if the dll
file is not where it thinks it is.
You may need to set reference path in your project properties or remove/add
the SharpDX reference to fix.

**Comment by blaisexen** [comment]
That's right too, nice to know Accord.Net users help each other.

**Comment by cesarsouza** [comment]
Hi @przemo1983,

After some testing, I've found that the issue you are experiencing happens because the framework was so far targeting SharpDX 2.6.x. Between 2.x and their new 3.x there were some significant changes, including the removal of the SharpDX.Bool struct that has popped up in your error message.

The framework was targeting 2.6 because this was the last version of SharpDX that was still compatible with the CLR 2 (and therefore C# 3.5). Since I would still like the framework to support 3.5, I hadn't updated SharpDX yet.

However, after some testing, I realized that it is possible to link to different versions of SharpDX using some conditions in the .csproj  project definition. So the next release will target SharpDX version 3.1.1 for the net40, net45 and net46 releases and SharpDX 2.6.3 for net35.

Sorry for the late response. So in short, if you would like to use the framework right now, you might have to revert back to an older version of SharpDX from the 2.6 branch.

Regards,
Cesar
